arch/cortex-m/samv71-hal/src/drivers/watchdog.rs

```diff
@@ -0,0 +1,155 @@
+//! Implementation of HAL Watchdog driver.
+//!
+//! Watchdog can be used to unconditionally reset or interrupt the MCU when program halts.
+//! To indicate that program is running, watchdog needs to be "fed" periodically.
+//! Minimal frequency of feeding can be adjusted by setting watchdog counter reset value (duration).
+//! Behavior of watchdog, it's state, and duration can be set using provided API.
+//!
+//! # Implementation remarks
+//! On SAMV71, Watchdog is enabled by default with duration of 16 seconds,
+//! and can only be configured ONCE. Consequent configurations will have no effect,
+//! until the MCU performs a hard reset (via reset controller or power cycle).
+
+pub mod watchdog_config;
+pub mod watchdog_error;
+
+use crate::pac::WDT;
+use fugit::MillisDurationU32;
+pub use watchdog_config::WatchdogConfig;
+pub use watchdog_error::WatchdogError;
+
+use self::watchdog_config::MAXIMUM_WATCHDOG_DURATION;
+
+/// Structure representing a watchdog.
+pub struct Watchdog {
+    /// Watchdog instance.
+    wdt: WDT,
+    /// Indicates whether the watchdog has already been configured (or disabled).
+    configured: bool,
+}
+
+/// # Safety
+/// Watchdog does not auto-implement Sync due to WDT structure containing a pointer.
+/// Since it owns WDT, and it's running in single-core environment, it's safe to share.
+unsafe impl Sync for Watchdog {}
+
+impl Watchdog {
+    /// Create a watchdog instance from PAC peripheral.
+    ///
+    /// # Arguments
```

Comment:
`Parameters` to be inline with the rest of the docs

Reply:
renamed in whole project